src/item/ItemComponentsTrait.php

```diff
@@ -65,10 +66,17 @@ public function getComponents(): CompoundTag {
 	 * Initializes the components and creates the base CompoundTag required for the components to be sent to a client.
 	 * This must be called before any properties or components are added otherwise it will break.
 	 */
-	protected function initComponent(string $texture, int $maxStackSize): void {
+	protected function initComponent(string $texture, int $maxStackSize, ?CreativeInventoryInfo $cti = null): void {
 		$this->componentTag = CompoundTag::create()
 			->setTag("components", CompoundTag::create()
 				->setTag("item_properties", CompoundTag::create()
+					->setInt("creative_category", match ($cti->getCategory()) {
+						CreativeInventoryInfo::CATEGORY_CONSTRUCTION => 1,
+						CreativeInventoryInfo::CATEGORY_NATURE => 2,
+						CreativeInventoryInfo::CATEGORY_EQUIPMENT => 3,
+						default => 4
+					})
+					->setString("creative_group", $cti->getGroup() ?? CreativeInventoryInfo::GROUP_MISC_FOOD)
```

Comment:
I'm not sure if we should be having any default behavior here. If creative category is not being set then we can assume they do not want it to be in creative? Why make things nullable?

Reply:
https://github.com/Steellgold/Customies/commit/20fac38cfbb6a956940599889d592e7afb7ed3dc